### PR TITLE
Improves error when axis is missing/incorrect attributes

### LINKED ISSUE #447: [Feature]: Improve `KeyError` message when `cf_xarray` can't find CF attributes for coordinates with `regrid2`
### What happened?

Spatial horizontal interpolation error for high resolution regional dataset

### What did you expect to happen?

I am interpolating high resolution regional data (over CONUS) to global grid, expecting data regridded with keeping missing area (thus, in regrided data, it will have data only over CONUS).

### Minimal Complete Verifiable Example

```python
>>> import xcdat as xc
>>> ds= xc.open_mfdataset("pr.historical.ACCESS-CM2.r1i1p1f1.mon.198101-200512.AC.v20230404.nc")
/global/homes/l/lee1043/.conda/envs/pmp_devel_20230213/lib/python3.9/site-packages/pandas/core/arrays/datetimelike.py:1384: CFWarning: this date/calendar/year zero convention is not supported by CF
  res_values = op(self.astype("O"), np.asarray(other))
>>> t_grid = xc.create_uniform_grid(-88.875, 88.625, 2.5, 0, 357.5, 2.5)
>>> ds_regrid = ds.regridder.horizontal('pr_tavg', t_grid, tool='regrid2')
```

### Relevant log output

```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/global/homes/l/lee1043/.conda/envs/pmp_devel_20230213/lib/python3.9/site-packages/xcdat/regridder/accessor.py", line 286, in horizontal
    output_ds = regridder.horizontal(data_var, self._ds)
  File "/global/homes/l/lee1043/.conda/envs/pmp_devel_20230213/lib/python3.9/site-packages/xcdat/regridder/regrid2.py", line 135, in horizontal
    output_axis_sizes = self._output_axis_sizes(input_data_var)
  File "/global/homes/l/lee1043/.conda/envs/pmp_devel_20230213/lib/python3.9/site-packages/xcdat/regridder/regrid2.py", line 175, in _output_axis_sizes
    axis_name = axis_name_map[standard_name]
KeyError: 'lat'
```

### Anything else we need to know?

Example input file for reproducing the error:
[pr.historical.ACCESS-CM2.r1i1p1f1.mon.198101-200512.AC.v20230404.nc.zip](https://github.com/xCDAT/xcdat/files/11204917/pr.historical.ACCESS-CM2.r1i1p1f1.mon.198101-200512.AC.v20230404.nc.zip)

### Environment

xcdat 0.4.0
pmp 3.0.1

### PR DESCRIPTION
**Pull request by jasonb5**
## Description

Improves regrid2 error when axis is missing from dataset or has incorrect attributes.

- Closes #447 

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

If applicable:

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass with my changes (locally and CI/CD build)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have noted that this is a breaking change for a major release (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by jasonb5** [comment]
I'd like to revisit this in the future and refactor regrid2 to use the newer xcdat axis functions to mitigate this issue all together.

File: tests/test_regrid.py [hunk 1]
```diff
@@ -157,6 +157,23 @@ def setup(self):
...
+
+        del ds.lat.attrs["axis"]
+
+        output_grid = grid.create_gaussian_grid(32)
+
+        regridder = regrid2.Regrid2Regridder(ds, output_grid)
+
+        with pytest.raises(
+            RuntimeError,
+            match="Could not find axis 'lat', ensure 'lat' exists and the attributes are correct.",
```
Comment (tomvothecoder) [inline tests/test_regrid.py:173]: I didn't know you could do a `match="..."` for the error. I'm going to start doing that too. Thanks for this.

File: xcdat/regridder/regrid2.py [hunk 2]
```diff
@@ -172,7 +172,12 @@ def _output_axis_sizes(self, da: xr.DataArray) -> Dict[str, int]:
         axis_name_map = {y[0]: x for x, y in da.cf.axes.items()}
 
         for standard_name in da.sizes.keys():
-            axis_name = axis_name_map[standard_name]
+            try:
+                axis_name = axis_name_map[standard_name]
+            except KeyError:
+                raise RuntimeError(
+                    f"Could not find axis {standard_name!r}, ensure {standard_name!r} exists and the attributes are correct."
+                )
```
Comment (tomvothecoder) [inline xcdat/regridder/regrid2.py:180]:
```suggestion
                raise RuntimeError(
                    f"Could not find axis {standard_name!r}, ensure {standard_name!r} "
                    "exists and the attributes are correct."
                )
```

`black` doesn't reformat string line length > limit (configured to 88) [yet](https://github.com/psf/black/issues/1802).

**Review by tomvothecoder — APPROVED** [review]
One minor code suggestion, then LGTM.

**Comment by tomvothecoder** [comment]
There's some weird `mamba` [error](https://github.com/xCDAT/xcdat/actions/runs/5214784066/jobs/9411561201?pr=481#step:7:14) going on in the GH Actions build. 

I'll take a look at it.

```bash
# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/usr/share/miniconda3/lib/python3.10/site-packages/conda/exceptions.py", line 1124, in __call__
        return func(*args, **kwargs)
      File "/usr/share/miniconda3/lib/python3.10/site-packages/conda_env/cli/main.py", line 78, in do_call
        exit_code = getattr(module, func_name)(args, parser)
      File "/usr/share/miniconda3/lib/python3.10/site-packages/conda/notices/core.py", line 109, in wrapper
        return func(*args, **kwargs)
      File "/usr/share/miniconda3/lib/python3.10/site-packages/conda_env/cli/main_update.py", line 132, in execute
        result[installer_type] = installer.install(prefix, specs, args, env)
      File "/usr/share/miniconda3/lib/python3.10/site-packages/mamba/mamba_env.py", line 55, in mamba_install
        index = load_channels(pool, channel_urls, repos, prepend=False)
      File "/usr/share/miniconda3/lib/python3.10/site-packages/mamba/utils.py", line 178, in load_channels
        repo = subdir.create_repo(pool)
    RuntimeError: Could not read JSON repodata file (/home/runner/conda_pkgs_dir/cache/47929eba.json) parse error line 593230: subdir mismatch

`$ /usr/share/miniconda3/condabin/mamba update -n xcdat_ci -f conda-env/ci.yml`
```

**Comment by jasonb5** [comment]
Nothing changed the `conda-env` files. I though it might be a caching issue so I restarted them last night. Guess that didn't clear it up, I might try restarting them again.

**Comment by jasonb5** [comment]
Guess this restart clear it up.

**Comment by tomvothecoder** [comment]
Yeah I tried restarting it a few times lol. You can merge if you are ready.

**Comment by lee1043** [comment]
I wonder if this improvement is going to be included for v0.6.0, which I hope so. Let me know if there is anything I can help on this PR.

**Comment by tomvothecoder** [comment]
> I wonder if this improvement is going to be included for v0.6.0, which I hope so. Let me know if there is anything I can help on this PR.

Yes this will be included in v0.6.0 since it is a simple fix. We're just waiting on @jasonb5 to merge whenever he's ready.

**Pull request merged**